cf/api/resources/service_keys_test.go

```diff
@@ -0,0 +1,72 @@
+package resources_test
+
+import (
+	"encoding/json"
+
+	. "github.com/cloudfoundry/cli/cf/api/resources"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ServiceKeyResource", func() {
+	var resource ServiceKeyResource
+
+	BeforeEach(func() {
+		err := json.Unmarshal([]byte(`
+    {
+      "metadata": {
+        "guid": "fake-service-key-guid",
+        "url": "/v2/service_keys/fake-guid",
+        "created_at": "2015-01-13T18:52:08+00:00",
+        "updated_at": null
+      },
+      "entity": {
+        "name": "fake-service-key-name",
+        "service_instance_guid":"fake-service-instance-guid",
+        "service_instance_url":"http://fake/service/instance/url",
+        "credentials": {
+          "username": "fake-username",
+          "password": "fake-password",
+          "host": "fake-host",
+          "port": 3306,
+          "database": "fake-db-name",
+          "uri": "mysql://fake-user:fake-password@fake-host:3306/fake-db-name"
+        }
+      }
+    }`), &resource)
+
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("Brokers supporting service keys", func() {
+		Describe("#ToFields", func() {
+			It("unmarshalls the fields of a service key resource", func() {
+				fields := resource.ToFields()
+
+				Expect(fields.Guid).To(Equal("fake-service-key-guid"))
+				Expect(fields.Name).To(Equal("fake-service-key-name"))
+			})
+		})
+
+		Describe("#ToModel", func() {
+			It("unmarshalls the service instance resource model", func() {
+				instance := resource.ToModel()
+
+				Expect(instance.Fields.Name).To(Equal("fake-service-key-name"))
+				Expect(instance.Fields.Guid).To(Equal("fake-service-key-guid"))
+				Expect(instance.Fields.Url).To(Equal("/v2/service_keys/fake-guid"))
+				Expect(instance.Fields.ServiceInstanceGuid).To(Equal("fake-service-instance-guid"))
+				Expect(instance.Fields.ServiceInstanceUrl).To(Equal("http://fake/service/instance/url"))
+
+				Expect(instance.Credentials).ToNot(BeNil())
```

Comment:
This line (assertion) is likely superfluous but OK to keep if all other similar tests have it.